examples/gno.land/r/demo/profile/handlers.gno

```diff
@@ -0,0 +1,99 @@
+package profile
+
+import (
+	"bytes"
+	"std"
+
+	"gno.land/p/demo/mux"
+	"gno.land/p/demo/ufmt"
+)
+
+const (
+	BaseURL           = "/r/demo/profile"
+	SetStringFieldURL = BaseURL + "?help&__func=SetStringField&field=%s"
+	SetIntFieldURL    = BaseURL + "?help&__func=SetIntField&field=%s"
+	SetBoolFieldURL   = BaseURL + "?help&__func=SetBoolField&field=%s"
+	ViewAllFieldsURL  = BaseURL + ":u/%s"
+	ViewFieldURL      = BaseURL + ":f/%s/%s"
+	PageSize          = 10 // Limit the number of addresses per page
+)
+
+func homeHandler(res *mux.ResponseWriter, req *mux.Request) {
+	var b bytes.Buffer
+
+	b.WriteString("## Setters\n")
+	for field := range stringFields {
+		link := ufmt.Sprintf(SetStringFieldURL, field)
+		b.WriteString(ufmt.Sprintf("- [Set %s](%s)\n", field, link))
+	}
+
+	for field := range intFields {
+		link := ufmt.Sprintf(SetIntFieldURL, field)
+		b.WriteString(ufmt.Sprintf("- [Set %s](%s)\n", field, link))
+	}
+
+	for field := range boolFields {
+		link := ufmt.Sprintf(SetBoolFieldURL, field)
+		b.WriteString(ufmt.Sprintf("- [Set %s Field](%s)\n", field, link))
+	}
+
+	b.WriteString("\n---\n\n")
+
+	res.Write(b.String())
+}
+
+func profileHandler(res *mux.ResponseWriter, req *mux.Request) {
+	var b bytes.Buffer
+	addr := req.GetVar("addr")
+
+	b.WriteString(ufmt.Sprintf("# Profile %s\n", addr))
+
+	address := std.Address(addr)
+
+	for field := range stringFields {
+		value := GetStringField(address, field, "n/a")
+		link := ufmt.Sprintf(SetStringFieldURL, field)
+		b.WriteString(ufmt.Sprintf("- %s: %s [Edit](%s)\n", field, value, link))
+	}
+
+	for field := range intFields {
+		value := GetIntField(address, field, 0)
+		link := ufmt.Sprintf(SetIntFieldURL, field)
+		b.WriteString(ufmt.Sprintf("- %s: %d [Edit](%s)\n", field, value, link))
+	}
+
+	for field := range boolFields {
+		value := GetBoolField(address, field, false)
+		link := ufmt.Sprintf(SetBoolFieldURL, field)
+		b.WriteString(ufmt.Sprintf("- %s: %t [Edit](%s)\n", field, value, link))
+	}
+
+	res.Write(b.String())
+}
+
+func fieldHandler(res *mux.ResponseWriter, req *mux.Request) {
+	var b bytes.Buffer
+	addr := req.GetVar("addr")
+	field := req.GetVar("field")
+
+	b.WriteString(ufmt.Sprintf("# Field %s for %s\n", field, addr))
+
+	address := std.Address(addr)
+	value := "n/a"
+	var editLink string
+
+	if _, ok := stringFields[field]; ok {
+		value = GetStringField(address, field, "n/a")
+		editLink = ufmt.Sprintf("%s&addr=%s", SetStringFieldURL, addr)
```

Comment:
would be nice to prefill the value too (`&value=%s`)

Reply:
https://github.com/gnolang/gno/pull/1983/commits/5789de857dbe5122bcf16bc5eaeaefd5d779a57c